flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/source/utils/BsonUtilsTest.java

```diff
@@ -0,0 +1,225 @@
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonDocument("fruit", new BsonString("apple")),
+                                new BsonDocument("fruit", new BsonString("banana")))
+                        < 0);
+
+        assertEquals(
+                0,
+                BsonUtils.compareBsonValue(
+                        new BsonDocument("fruit", new BsonString("banana")),
+                        new BsonDocument("fruit", new BsonString("banana"))));
+
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonDocument("fruit", new BsonString("cherry")),
+                                new BsonDocument("fruit", new BsonString("banana")))
+                        > 0);
+
+        // test compare Timestamp
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonTimestamp(1600000000), new BsonTimestamp(1700000000))
+                        < 0);
+
+        assertEquals(
+                0,
+                BsonUtils.compareBsonValue(
+                        new BsonTimestamp(1700000000), new BsonTimestamp(1700000000)));
+
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonTimestamp(1800000000), new BsonTimestamp(1700000000))
+                        > 0);
+
+        // test compare RegEx
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonRegularExpression("[a-xA-X]"),
+                                new BsonRegularExpression("[b-yB-Y]"))
+                        < 0);
+
+        assertEquals(
+                0,
+                BsonUtils.compareBsonValue(
+                        new BsonRegularExpression("[b-yB-Y]"),
+                        new BsonRegularExpression("[b-yB-Y]")));
+
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonRegularExpression("[c-zC-Z]"),
+                                new BsonRegularExpression("[b-yB-Y]"))
+                        > 0);
+
+        // test compare JavaScript
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonJavaScriptWithScope(
+                                        "console.log('apple');", new BsonDocument()),
+                                new BsonJavaScriptWithScope(
+                                        "console.log('banana');", new BsonDocument()))
+                        < 0);
+
+        assertEquals(
+                0,
+                BsonUtils.compareBsonValue(
+                        new BsonJavaScriptWithScope("console.log('banana');", new BsonDocument()),
+                        new BsonJavaScriptWithScope("console.log('banana');", new BsonDocument())));
+
+        assertTrue(
+                BsonUtils.compareBsonValue(
+                                new BsonJavaScriptWithScope(
```

Comment:
It will be better to  test two JavascriptWithScope with same codes but different scope